tests/test__collection_api.py

```diff
@@ -2180,6 +2180,69 @@ def test__find_or_and(self):
         ])
         self.assertEqual([2], [d['x'] for d in self.db.collection.find(search_filter)])
 
+    def test__aggregate_replace_root(self):
+        self.db.a.insert_many([
+            {'_id': 1, 'pets': {'dogs': 2, 'cats': 3}},
+            {'_id': 2, 'pets': {'hamsters': 3, 'cats': 4}}
+        ])
+        actual = self.db.a.aggregate([
+            {'$replaceRoot': {'newRoot': '$pets'}}
+        ])
+        self.assertListEqual([
+            {'dogs': 2, 'cats': 3},
+            {'hamsters': 3, 'cats': 4}
+        ], list(actual))
+
+    def test__aggregate_replace_root_non_existing(self):
+        self.db.a.insert_many([
+            {'_id': 1, 'pets': {'dogs': 2, 'cats': 3}},
+            {'_id': 2, 'pets': {'hamsters': 3, 'cats': 4}}
+        ])
+        with self.assertRaises(mongomock.OperationFailure) as err:
+            self.db.a.aggregate([
+                {'$replaceRoot': {
+                    'newRoot': '$not_here'
+                }}
+            ])
+        self.assertEqual(
+            'Could not complete $replaceRoot with given expression.',
```

Comment:
This is not the error I get when I run your example against a real Mongo server. I get:
```
pymongo.errors.OperationFailure: 'newRoot' expression must evaluate to an object, but resulting value was: MISSING. Type of resulting value: 'missing'. Input document: {}
```

Reply:
Do we need to write the exact error the mongo server writes? In the other locations in te code the errors were written differently. Doesn't it only matter that it will fail in the same places?

Reply:
I'm OK not to have the same text message. However it should be the same kind of exception (OperationFailure) and explain to the user what went wrong. Here I like the reminder that it must evaluate to an object.